src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1376,6 +1376,9 @@ private function getServiceCall($id, Reference $reference = null)
             return '$this';
         }
 
+        if ($this->container->hasDefinition($id) && !$this->container->getDefinition($id)->isPublic()) {
+            return "\$this->services[(isset(\$this->services['$id']) || \$this->{$this->generateMethodName($id)}()) && false ?: '$id']";
```

Comment:
Good catch, fixed. You'll scream when looking at the generated code, yet it works well and I didn't find any better syntax that worked on 5.5 :)